src/streaming-ingestion/flink-etl/src/main/java/software/aws/solution/clickstream/plugin/enrich/IPEnrichment.java

```diff
@@ -0,0 +1,157 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+package software.aws.solution.clickstream.plugin.enrich;
+
+import com.maxmind.db.CHMCache;
+import com.maxmind.db.MaxMindDbConstructor;
+import com.maxmind.db.MaxMindDbParameter;
+import com.maxmind.db.Reader;
+import lombok.Getter;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+
+import java.io.ByteArrayInputStream;
+import java.net.InetAddress;
+import java.util.Map;
+import java.util.Optional;
+
+public class IPEnrichment implements Enrichment {
+    public static final String PARAM_KEY_IP = "ip";
+    public static final String PARAM_KEY_LOCALE = "locale";
+    public static final String PARAM_KEY_BUCKET = "geoBucketName";
+    public static final String PARAM_KEY_FILE_NAME = "geoFileKey";
+    private static final Logger LOG = LoggerFactory.getLogger(IPEnrichment.class);
+    public ObjectNode enrich(ObjectNode geoNode, Map<String,String> paramMap) {
+        String ip = paramMap.get(PARAM_KEY_IP);
+        String locale = paramMap.get(PARAM_KEY_LOCALE);
+        String bucket = paramMap.get(PARAM_KEY_BUCKET);
+        String fileName = paramMap.get(PARAM_KEY_FILE_NAME);
+        LOG.info("IPEnrichment transform begin enrich.");
+        S3Client s3 = S3Client.builder().build();
+        try {
+            GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                    .bucket(bucket)
+                    .key(fileName)
+                    .build();
+            ResponseBytes<GetObjectResponse> objectBytes = s3.getObjectAsBytes(getObjectRequest);
+            try (Reader reader = new Reader(
```

Comment:
This `try...catch...` looks unnecessary.

Reply:
done